pkg/scheduler/scheduler.go

```diff
@@ -178,8 +179,8 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 			e.inadmissibleReason = fmt.Sprintf("Could not obtain workload namespace: %v", err)
 		} else if !cq.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
 			e.inadmissibleReason = "Workload namespace doesn't match ClusterQueue selector"
-		} else if !e.assignFlavors(log, snap.ResourceFlavors, cq) {
-			e.inadmissibleReason = "Workload didn't fit in the remaining quota"
+		} else if ok, err := e.assignFlavors(log, snap.ResourceFlavors, cq); !ok {
+			e.inadmissibleReason = fmt.Sprintf("Workload didn't fit: %s", err.Error())
```

Comment:
What if the message is too long for the Event? Make sure to trim it:

https://github.com/kubernetes/kubernetes/blob/f02682c628c530219966a00ae002d799f0d813dc/pkg/scheduler/schedule_one.go#L836-L844